scylla-cdc/src/cdc_reader.rs

```diff
@@ -0,0 +1,213 @@
+use std::sync::Arc;
+
+use anyhow;
+use chrono::Duration;
+use core::time;
+use futures::future::RemoteHandle;
+use futures::stream::{FusedStream, FuturesUnordered, StreamExt};
+use futures::FutureExt;
+use scylla::Session;
+use tokio::task::JoinError;
+
+use crate::cdc_types::GenerationTimestamp;
+use crate::consumer::ConsumerFactory;
+use crate::reader::StreamReader;
+use crate::stream_generations::GenerationFetcher;
+
+pub struct CDCReader {
+    // Tells the worker to stop
+    // Usage of the "watch" channel will make it possible to change the the timestamp later,
+    // for example if somebody loses patience and wants to stop now not later
+    end_timestamp: tokio::sync::watch::Sender<Duration>,
+}
```

Comment:
I don't like the name `Reader`, because it may suggest that it's something similar to `StreamReader`, but I couldn't come up with a better name - I'm open for suggestions.

Reply:
It reads a whole CDC log, so maybe `CDCLogReader`?